inc/saivlan.h

```diff
@@ -98,6 +98,12 @@ typedef enum _sai_vlan_attr_t
       * Value Range SAI_SWITCH_ATTR_VLAN_USER_META_DATA_RANGE */
     SAI_VLAN_ATTR_META_DATA,
 
+    /**
+      * The Multicast 802.1BR E-Channel Id (ECID), to be used in the flooding traffic
+      * sent by the CB to Port Extenders (PE). [sai_uint32_t] (CREATE_AND_SET) (default to 0)
+      */
+    SAI_VLAN_ATTR_FLOODING_ECID,
```

Comment:
Dell - Internal Use - Confidential
Hi Min Yao,

I will address this.

The replication in CB can be achieved using the Multicast object that is being discussed in OCP. Will address this after the Multicast Pull request gets merged.

The dot1br header file pull request is an old one that needs update. Will take care of this then.

Regards,
Ravi

From: Min Yao [mailto:notifications@github.com]
Sent: Wednesday, August 31, 2016 9:29 PM
To: opencomputeproject/SAI <SAI@noreply.github.com>
Cc: Sivasankar, Ravikumar <Ravikumar_Sivasankar@Dell.com>; Mention <mention@noreply.github.com>
Subject: Re: [opencomputeproject/SAI] IEEE 802.1BR SAI proposal. (#73)


In inc/saivlan.h<https://github.com/opencomputeproject/SAI/pull/73#discussion_r77015933>:

> @@ -98,6 +98,12 @@ typedef enum _sai_vlan_attr_t

>        * Value Range SAI_SWITCH_ATTR_VLAN_USER_META_DATA_RANGE */

>      SAI_VLAN_ATTR_META_DATA,

>

> +    /**

> +      * The Multicast 802.1BR E-Channel Id (ECID), to be used in the flooding traffic

> +      * sent by the CB to Port Extenders (PE). [sai_uint32_t] (CREATE_AND_SET) (default to 0)

> +      */

> +    SAI_VLAN_ATTR_FLOODING_ECID,

@rsivasankar<https://github.com/rsivasankar> Now we're on the same page. : )

"each Vlan must be associated with a list of {Cascading Port, ECID} tuple" -- this is exactly what I mean. My opinion is that we should allow the possibility that ControlStack allocates same ECID for two E-Channel in different Internal Bridge Port Extenders within one CB, to keep conformance to the IEEE standard.
I remember that DELL has mentioned that only CB replication is required for now. In this case, this SAI_VLAN_ATTR_FLOODING_ECID attribute is not needed. Multicast ECID is only need when doing PE replication. I'm OK that DELL allocates different ECID for every ExtendedPort but I'm not OK with this SAI_VLAN_ATTR_FLOODING_ECID attribute.

—
You are receiving this because you were mentioned.
Reply to this email directly, view it on GitHub<https://github.com/opencomputeproject/SAI/pull/73/files/279f0335c33c0c831ec955e1cf93daf61adf52d5#r77015933>, or mute the thread<https://github.com/notifications/unsubscribe-auth/AOoHu9aNAi1fNPcpgJW_Fu0zd4KbgdZSks5qlaSngaJpZM4HpSGE>.